lib/filewatch/discover.rb

```diff
@@ -0,0 +1,141 @@
+require_relative 'watched_file'
+
+module FileWatch
+  class Discover
+    attr_reader  :logger, :files, :wf_vars
+
+    def initialize(opts, loggr)
+      @logger = loggr
+      @watching = []
+      @exclude = []
+      @files = {}
+      set_ignore_older(opts[:ignore_older])
+      set_close_older(opts[:close_older])
+      @wf_vars = [opts[:delimiter], @ignore_older, @close_older]
+      exclude(opts[:exclude])
+    end
+
+    def logger=(loggr)
+      @logger = loggr
+      @converter.logger = loggr
+    end
+
+    def add_converter(converter)
```

Comment:
I personnaly dont use fluent methods that much, unless I write a DSL.

This could be valid too.

```ruby
Foo.converters << Converter
```